schemas/1.19.0

```
@@ -0,0 +1,73 @@
+file_format: 1.1.0
+schema_url: https://opentelemetry.io/schemas/1.19.0
+versions:
+  1.19.0:
+    spans:
+      changes:
+        # https://github.com/open-telemetry/opentelemetry-specification/pull/3190
+        - rename_attributes:
+            attribute_map:
+              http.user_agent: user_agent.original
+    Resources:
```

I have no idea why this has to be uppercase `Resources`, but schema validation fails with lowercase `resources`

---

Weird... it _seems_ the error comes from [build tools](https://github.com/open-telemetry/build-tools/blob/main/schemas/main.go#L30) which _seems_ to use [OTel Go schema parsing](https://github.com/open-telemetry/opentelemetry-go/blob/schema/v0.0.3/schema/v1.1/parser.go#L46) which reports this as a "simple" decoding problem.

Is `resources` a reserved keyword perhaps? Maybe @MrAlias @MadVikingGod know?

---

Pretty sure it comes from this: https://github.com/open-telemetry/opentelemetry-go/blob/59daa05b19aa49f90f4b833871a89f3329b40872/schema/v1.1/ast/ast_schema.go#L45

There is no field tag specifying an alternate yaml identified so it uses the default field name "Resource". That could be updated to accept lowercase if needed (it would break existing uppercase though)

---

do u know why `spans` (lowercase) works currently, even though it's uppercase `Spans` in the linked code similar to `Resources`?

---

Right. I think it's because it is the field type as well. I'd need to experiment to be sure though.

---

There seems to be something more complex than I was thinking going on here. Our test data uses a lowercase `resource` and all our tests pass.

https://github.com/open-telemetry/opentelemetry-go/blob/59daa05b19aa49f90f4b833871a89f3329b40872/schema/v1.1/testdata/valid-example.yaml#L50

---

Looking into this further, I think by switching to `Resources` you are entirely ignoring this section. I haven't verified this, but I did find the answer to why the original parsing was failing.

Originally you structured the `resource` section like this:

```yaml
    resources:
      changes:
        # https://github.com/open-telemetry/opentelemetry-specification/pull/3190
        - rename_attributes:
            attribute_map:
              browser.user_agent: user_agent.original
```

But the AST for the resource section doesn't have a "attribute_map" field like the spans do. If you updated to the following it should work:

```yaml
    resources:
      changes:
        # https://github.com/open-telemetry/opentelemetry-specification/pull/3190
        - rename_attributes:
          browser.user_agent: user_agent.original
```

---

cc @tigrannajaryan

---

> Looking into this further, I think by switching to `Resources` you are entirely ignoring this section. I haven't verified this ...

Ah, it does indeed look to silently drop the `Resources` section: https://github.com/go-yaml/yaml/issues/136

I think we may need to enable this: https://pkg.go.dev/gopkg.in/yaml.v3#Decoder.KnownFields

---

https://github.com/open-telemetry/opentelemetry-go/issues/3742